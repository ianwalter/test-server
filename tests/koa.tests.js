const { test } = require('@ianwalter/bff')
const { requester } = require('@ianwalter/requester')
const { createKoaServer } = require('..')

test('Koa server created', async ({ expect }) => {
  const server = await createKoaServer()
  expect(server.port).toBeGreaterThan(0)
  expect(server.url).toBeTruthy()
  await server.close()
})

test('Koa request handler', async ({ expect }) => {
  const server = await createKoaServer()
  const msg = 'Nobody Lost, Nobody Found'
  server.use(ctx => (ctx.body = msg))
  const { body } = await requester.get(server.url)
  expect(body).toBe(msg)
  await server.close()
})

test('Koa json response', async ({ expect }) => {
  const server = await createKoaServer()
  server.use(ctx => (ctx.body = { name: 'Out There On the Ice' }))
  const { body } = await requester.get(server.url)
  expect(body).toMatchSnapshot()
  await server.close()
})

test('Koa json request', async ({ expect }) => {
  const server = await createKoaServer()
  server.use(ctx => (ctx.body = ctx.request.body))
  const body = { name: 'When Am I Gonna Lose You' }
  const response = await requester.post(server.url, { body })
  expect(response.body).toEqual(body)
  await server.close()
})

test.skip('Koa cors', async (t, page) => {
  const server = await createKoaServer()
  server.use(ctx => (ctx.body = 'Moments'))
  const result = await page.evaluate(
    url => window.fetch(url).then(response => response.text()),
    server.url
  )
  t.is(result, 'Moments')
})

test('Koa error', async ({ pass }) => {
  const server = await createKoaServer()
  server.use(() => new Promise((resolve, reject) => reject(new Error('Nooo!'))))
  try {
    await requester.get(server.url)
  } catch (err) {
    pass()
  } finally {
    await server.close()
  }
})
