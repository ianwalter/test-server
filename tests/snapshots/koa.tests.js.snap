// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`json response 1`] = `
Object {
  "name": "Out There On the Ice",
}
`;
